cpp/src/utils.cpp

```diff
@@ -305,4 +354,42 @@ size_t linearize(const legate::DomainPoint& lo,
   return legate::dim_dispatch(point.dim, linearize_fn{}, lo, hi, point);
 }
 
+std::vector<uint8_t> serialize_arrow_type(std::shared_ptr<arrow::DataType> type)
+{
+  auto schema = arrow::schema({arrow::field("type", type)});
+  auto buffer = ARROW_RESULT(arrow::ipc::SerializeSchema(*schema));
+  return std::vector<uint8_t>(buffer->data(), buffer->data() + buffer->size());
+}
+
+std::vector<uint8_t> serialize_arrow_types(std::vector<std::shared_ptr<arrow::DataType>> types)
+{
+  std::vector<std::shared_ptr<arrow::Field>> fields;
+  for (const auto& type : types) {
+    fields.push_back(arrow::field(std::to_string(fields.size()), type));
+  }
+  auto schema = arrow::schema(fields);
+  auto buffer = ARROW_RESULT(arrow::ipc::SerializeSchema(*schema));
+  return std::vector<uint8_t>(buffer->data(), buffer->data() + buffer->size());
+}
+
+std::shared_ptr<arrow::DataType> deserialize_arrow_type(const std::vector<uint8_t>& data)
+{
+  auto buffer       = arrow::Buffer::Wrap(data.data(), data.size());
+  auto input_stream = arrow::io::BufferReader(buffer);
+  auto schema       = ARROW_RESULT(arrow::ipc::ReadSchema(&input_stream, nullptr));
+  return schema->fields().at(0)->type();
+}
+
+std::vector<std::shared_ptr<arrow::DataType>> deserialize_arrow_types(
+  const std::vector<uint8_t>& data)
+{
+  auto buffer       = arrow::Buffer::Wrap(data.data(), data.size());
+  auto input_stream = arrow::io::BufferReader(buffer);
+  auto schema       = ARROW_RESULT(arrow::ipc::ReadSchema(&input_stream, nullptr));
+  std::vector<std::shared_ptr<arrow::DataType>> types;
+  for (const auto& field : schema->fields()) {
+    types.push_back(field->type());
+  }
+  return types;
```

Comment:
I really like this serialization!

However, one (hopefully small) ask is that I think it would be nice to add overloads to `task_argument.cpp/hpp` (some stuff there are inline versions, so for simplicity maybe `.hpp`).

Basically, I would like to be able to write:
```
auto arrow_type = argument::get_next_scalar<arrow::DataType>(ctx));
```
(or `std::shared_ptr<arrow::DataType>`) and also the same for `argument::add_next_scalar(arrow_type)` to just work.

The code here can stay, I just think it would be nice to not have to call the serialize/deserialize explicitly anyway.